uamqp/management_link.py

```diff
@@ -137,16 +143,30 @@ def _on_message_received(self, message):
             )
             self._pending_operations.remove(to_remove_operation)
 
-    def _on_send_complete(self, message, reason, state):
-        if SEND_DISPOSITION_REJECT in state:  # either rejected or accepted
+    def _on_send_complete(self, message_delivery, reason, state):  # todo: reason is never used, should check spec
+        if SEND_DISPOSITION_REJECT in state:
             to_remove_operation = None
             for operation in self._pending_operations:
-                if message == operation.message:
+                if message_delivery.message == operation.message:
                     to_remove_operation = operation
                     break
             self._pending_operations.remove(to_remove_operation)
-            to_remove_operation.on_execute_operation_complete(
-                ManagementExecuteOperationResult.ERROR, None, None, message)
+            if SEND_DISPOSITION_REJECT in state:  # either rejected or accepted
+                # TODO: better error handling
+                # The callback is defined in management_operation.py
+                # TODO: AMQPException is too general? to be more specific: MessageReject(Error) or AMQPManagementError?
+                #  or should there an error mapping which maps the condition to the error type
+                to_remove_operation.on_execute_operation_complete(
+                    ManagementExecuteOperationResult.ERROR,
+                    None,
+                    None,
+                    message_delivery.message,
+                    error=AMQPException(
+                        state[SEND_DISPOSITION_REJECT][0][0],
+                        state[SEND_DISPOSITION_REJECT][0][1],
+                        state[SEND_DISPOSITION_REJECT][0][2],
```

Comment:
Nitpick: Can we add a quick # Comment for each line so we can quickly see what each of these indexes represent?

Reply:
Given that we're using Tuples everywhere for performance - I've tried to be super verbose about documenting the indexes with inline-comments to make maintenance easier.

Reply:
this is a good suggestion, comment added, will keep this in mind